aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -104,7 +112,15 @@ pub(crate) fn run_script_prologue(
             MoveValue::U64(txn_expiration_timestamp_secs),
             MoveValue::U8(chain_id.id()),
         ];
-        (&APTOS_TRANSACTION_VALIDATION.fee_payer_prologue_name, args)
+        if required_deposit.is_some() {
+            args.push(required_deposit.as_move_value());
```

Comment:
I don't think this works since the binary and framework are not updated at the same time, we need to handle the case that the framework is not updated

Reply:
fixed by gating randomness deposit calculation with randomness feature flags (and in the future. feature X deposit calculation with X feature flags)
?